### DISCUSSION TITLE
Help how to use SVGImage from cs in c# wpf

### DISCUSSION THREAD

**Discussion by hpvcodeDev** [dotnetprojects/SVGImage#86] [Q&A]
Hello, I am using SVGImage and for example to add an image to my button from my xaml file I use it this way

`<Image Source="{svg1:SvgIcon AppName=Example, Source=/Imagen/Open/onedrive.svg}"  Width="29" Stretch="Uniform" Margin=" 10 3 3 3" />
`
and it works great for me, but now I am creating a button from the cs and I wanted to know what using will be used as a reference and what the call method would be to assign a sav image to my button from the cs.

**Comment by paulushub** [comment]
`SvgIcon` and `SvgIconConverter` are markup extensions not controls, which means they are basically used in XAML.
`SvgIconConverter`, however, supported binding, which means you can bind to it in C# source codes.

If you define the button in XAML or the button has XAML part, then you can also use any of the above. Other than that consider using `SVGImage` control or creating a new control and rendering the SVG as in `SvgIcon`.  The `SvgIcon` uses `FileSvgReader` internally (because it was ported from SharpVectors), you can copy it if needed for your use case.

Here is an implementation of buttons for references: https://github.com/Gedd18/swui
